core/src/main/java/net/adoptopenjdk/icedteaweb/userdecision/FileStoreEntry.java

```diff
@@ -1,54 +1,34 @@
 package net.adoptopenjdk.icedteaweb.userdecision;
 
-import net.adoptopenjdk.icedteaweb.Assert;
-
 import java.net.URL;
-import java.util.Objects;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
 import java.util.Set;
 
-public class FileStoreEntry {
-    private UserDecision.Key userDecisionKey;
-    private String userDecisionValue;
-
-    private URL domain;
-    private Set<String> jarNames;
+class FileStoreEntry {
+    private final URL domain;
+    private final Set<String> jarNames;
+    private final Map<UserDecision.Key, String> decisions = new HashMap<>();
```

Comment:
This is the main change.
A `FileStoreEntry` has now a map of `UserDecision.Key -> String`